crates/autopilot/src/run_loop.rs

```diff
@@ -513,6 +517,17 @@ pub fn solve_request(
                             .collect()
                     };
                 let order_is_untouched = remaining_order.executed_amount.is_zero();
+
+                let fee_policies = match order.metadata.class {
+                    OrderClass::Market => vec![],
+                    OrderClass::Liquidity => vec![],
+                    // todo https://github.com/cowprotocol/services/issues/2092
+                    // skip protocol fee for limit orders with in-market price
+
+                    // todo https://github.com/cowprotocol/services/issues/2115
+                    // skip protocol fee for TWAP limit orders
+                    OrderClass::Limit(_) => vec![fee_policy_to_domain(&fee_policy)],
```

Comment:
Shouldn't this somehow take `fee_policy_skip_market_orders` into account and include the match statement on `order.metadata.class`?

I'd create a new protocol_fee module and have a struct apply the fee. Note that in the future we will likely have logic that based on app_data adds additional fees here.

Reply:
This will be implemented as a separate task: https://github.com/cowprotocol/services/issues/2092

Let's also move this logic into a separate file as part of ☝️ task, to unblock this PR and merge it.